qiskit/aqua/operators/list_ops/summed_op.py

```diff
@@ -116,6 +116,7 @@ def simplify(self) -> 'SummedOp':
     # Try collapsing list or trees of Sums.
     # TODO be smarter about the fact that any two ops in oplist could be evaluated for sum.
     def reduce(self) -> OperatorBase:
+        self.oplist = chop_summands()
```

Comment:
So one aspect to bear in mind is that we are creating new SummedOp instances when operating on the oplist not modifying self. So for instance one would build out a new oplist that has the chopped coeffs, and this new oplist will end up being used to created the resultant chopped/simplified operator.

Reply:
Do I want to do something like `chop_ops = chop_summands()` and then use chop_ops in place of `self.oplist` throughout the rest of the function?

Reply:
If you look at what its doing for reduce the list comprehension where its dealing with reduce for each op assigns the new list to a local variable reduced_ops. For sure it should not modify self.oplist so using some local var seems like the thing to do. I am not sure that here is the 'right' place to have this. Maybe chop should be available as an instance method that will simply remove elements that are below the threshold. The method ought to take a threshold. Maybe it should be done at the end of simplify where we have combined operators etc so that 2 or more may combine so the coeff is below the threshold but individually not. I am not so sure of just having it there because then arguably simplify ought to have a threshold too. My feeling is that someone wanting the operator chopped should do it explicitly. An automatic chop could remove operators whose coeffs were exactly zero I guess as part of simplify.